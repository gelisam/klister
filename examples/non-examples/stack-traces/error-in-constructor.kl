#lang "prelude.kl"

(import "lambda-case.kl")

(datatype (Pair A B)
  (pair A B))

(define fst
  (lambda-case
    [(pair x _) x]))

(define snd
  (lambda-case
    [(pair _ y) y]))

(define fail (lambda (thing)
               (let (go (pair thing (error 'Im-an-error)))
                 go)))

(example (fail 23))
