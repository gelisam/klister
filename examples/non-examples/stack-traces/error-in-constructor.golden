In Phase: p0:
    error-in-constructor.kl:17.45-17.56: Im-an-error
    stack trace:
    ---- in constructor pair
                        in field 2
    ---- in let #[error-in-constructor.kl:17.22-17.24]<go>
    ---- Halt
